### PR TITLE
Enable `--all` to uninstall all managed Pythons

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Allows `--all` as an alternative to specifying specific targets.

## Test Plan

Verified that `cargo run python uninstall` still fails.

```
❯ cargo run python uninstall --all
   Compiling uv-cli v0.0.1 (/Users/crmarsh/workspace/puffin/crates/uv-cli)
   Compiling uv v0.2.23 (/Users/crmarsh/workspace/puffin/crates/uv)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 2.86s
     Running `target/debug/uv python uninstall --all`
warning: `uv python uninstall` is experimental and may change without warning.
Searching for Python installations
Found existing installation: cpython-3.9.18-macos-aarch64-none
Found existing installation: cpython-3.8.18-macos-aarch64-none
Found existing installation: cpython-3.8.12-macos-aarch64-none
Found existing installation: cpython-3.12.1-macos-aarch64-none
Found existing installation: cpython-3.11.7-macos-aarch64-none
Found existing installation: cpython-3.10.13-macos-aarch64-none
Uninstalled cpython-3.10.13-macos-aarch64-none
Uninstalled cpython-3.11.7-macos-aarch64-none
Uninstalled cpython-3.12.1-macos-aarch64-none
Uninstalled cpython-3.8.12-macos-aarch64-none
Uninstalled cpython-3.8.18-macos-aarch64-none
Uninstalled cpython-3.9.18-macos-aarch64-none
Uninstalled 6 versions in 479ms
```

### CODE REVIEW

File: crates/uv/src/commands/python/uninstall.rs [hunk 1]
```diff
@@ -74,14 +92,10 @@ pub(crate) async fn uninstall(
...
-            writeln!(
-                printer.stderr(),
-                "No Python installations found matching the requests"
-            )?;
-        }
+        writeln!(
+            printer.stderr(),
+            "No Python installations found matching the requests"
+        )?;
         return Ok(ExitStatus::Failure);
```
Comment (zanieb) [inline crates/uv/src/commands/python/uninstall.rs:99]: Why this change?

Comment (zanieb) [reply crates/uv/src/commands/python/uninstall.rs:99]: Oh I see nevermind

File: crates/uv/src/settings.rs [hunk 2]
```diff
@@ -376,6 +376,7 @@ impl PythonInstallSettings {
 #[derive(Debug, Clone)]
 pub(crate) struct PythonUninstallSettings {
     pub(crate) targets: Vec<String>,
+    pub(crate) all: bool,
```
Comment (zanieb) [inline crates/uv/src/settings.rs:379]: We should mark these as conflicting right?

Comment (zanieb) [reply crates/uv/src/settings.rs:379]: Omg sorry I'm dumb I looked at the settings struct. Bad review comments...

**Review by zanieb — APPROVED** [review]

File: crates/uv/src/settings.rs (hunk 2 above)
Comment (charliermarsh) [reply crates/uv/src/settings.rs:379]: Lol appreciate you!
